File: 162717909/cash/cash.c
#include <stdio.h>
void greedy(int *a, int *b, int g);
int main()
{
    int owed;
    do
    {
        printf("Change owed: ");
        scanf("%d", &owed);
    }
    while (owed <= 0);

    int num = 0;
    int *a = &owed;
    int *b = &num;
    greedy(a, b, 25);
    greedy(a, b, 10);
    greedy(a, b, 5);
    greedy(a, b, 1);
    printf("%d\n", num);
    return 0;
}
void greedy(int *a, int *b, int g)
{
    if (*a >= g)
    {
        while (*a >= g)
        {
            *a -= g;
            (*b)++;
        }
    }
}

File: 162717909/sentimental-cash/cash.py
from cs50 import get_float
while True:
    cash = get_float("Change: ")
    if 0 < cash:
        break

cash = int(cash*100)
num = 0
print(cash)

while cash >= 25:
    cash -= 25.
    num += 1
while cash >= 10:
    cash -= 10
    num += 1
while cash >= 5:
    cash -= 5
    num += 1
while cash > 0:
    cash -= 1
    num += 1
print(f"{num}")
